### PR TITLE
Perl prove goodie json

### PR DESCRIPTION
**Pull request by dnmfarrell**
**What does your Instant Answer do?**

Show the command line args for `prove`

**What is the data source for your Instant Answer? (Provide a link if possible)**

The official docs: http://perldoc.perl.org/prove

**Why did you choose this data source?**

It's canonical

**Are there any other alternative (better) data sources?**

No

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**

Perl developers, mostly

**Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?**

No

**Are you having any problems? Do you need our help with anything?**

**Just feedback on the styling, does it look OK, maybe another template type would be better suited?**

**Where did you hear about DuckDuckHack? (For first time contributors)**

N/A

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

screenshot from 2015-08-29 14-47-27

-----
IA Page: https://duck.co/ia/view/prove_cheat_sheet

### CODE REVIEW

**Comment by dnmfarrell** [comment]
What does your Instant Answer do?

Show the command line args for `prove`

What is the data source for your Instant Answer? (Provide a link if possible)

The official docs: http://perldoc.perl.org/prove

Why did you choose this data source?

It's canonical

Are there any other alternative (better) data sources?

No

Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)

Perl developers, mostly

Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?

No

Are you having any problems? Do you need our help with anything?

Just feedback on the styling, does it look OK, maybe another template type would be better suited?

Where did you hear about DuckDuckHack? (For first time contributors)

N/A

What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)

screenshot from 2015-08-29 14-47-27

File: share/goodie/cheat_sheets/json/prove.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{
+    "id": "prove_cheat_sheet",
+    "name": "prove",
+    "description": "Perl test runner for the Test Anything Protocol",
+    "metadata": {
+        "sourceName": "prove Manual",
+        "sourceUrl": "http://perldoc.perl.org/prove.html"
+    },
+    "template_type": "keyboard",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/prove.json]: @dnmfarrell if you switch this to `terminal` then it'll change the UI and you won't have to use `[]` to create code blocks :)

File: share/goodie/cheat_sheets/json/prove.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{
+    "id": "prove_cheat_sheet",
+    "name": "prove",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/prove.json]: Let's go with "Prove" or maybe "Perl Prove" ?

**Comment by dnmfarrell** [comment]
Would "Perl prove" be OK? The name of the program is [prove](http://perldoc.perl.org/prove.html)

**Comment by moollaza** [comment]
> Would "Perl prove" be OK? The name of the program is prove

:+1:

File: share/goodie/cheat_sheets/json/prove.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    "metadata": {
+        "sourceName": "prove Manual",
+        "sourceUrl": "http://perldoc.perl.org/prove.html"
+    },
+    "template_type": "keyboard",
+    "section_order": ["Usage", "Options"],
+    "sections": {
+        "Usage": [
+        {
+            "key": "[prove \\[option\\] \\[filepath\\] ]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/prove.json]: You can remove the `[]` around the whole query here.

File: share/goodie/cheat_sheets/json/prove.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
+{
+    "id": "prove_cheat_sheet",
+    "name": "prove",
+    "description": "Perl test runner for the Test Anything Protocol",
+    "metadata": {
+        "sourceName": "prove Manual",
+        "sourceUrl": "http://perldoc.perl.org/prove.html"
+    },
+    "template_type": "keyboard",
+    "section_order": ["Usage", "Options"],
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/prove.json]: I think we should switch the order of these. Can we add an "Example" section with a few common examples?

- `prove -Ilib`
- `prove -Ilib t/MyTest.t`
- `prove -Ilib t/subdir/`
etc,

**Comment by moollaza** [comment]
@dnmfarrell thanks again for contributing, we really appreciate it! I've left you some feedback that needs to be implemented. Once things are updated please let me so so I can review this again. If you have any questions, feel free to ask them here!

**Comment by dnmfarrell** [comment]
Thanks for the feedback, love the idea of an examples section. I've rebased and added an examples section. See what you think, maybe we need fewer/more/different examples?

**Comment by mintsoft** [comment]
@dnmfarrell This diff seems huge now; have you rebased?

**Comment by moollaza** [comment]
@dnmfarrell apologies for the delay here and problems. I think it wold be best to create a new PR for this one, from a fresh branch. That will be the simplest way to get this one merged!

**Pull request closed without merging**